trieste/space.py

```diff
@@ -152,42 +152,56 @@ def __mul__(self, other: DiscreteSearchSpace) -> DiscreteSearchSpace:
         return DiscreteSearchSpace(tf.reshape(cartesian_product, [N * M, -1]))
 
 
+Vector = TypeVar("Vector", tf.Tensor, List[float])
+""" A type variable representing either a :class:`tf.Tensor` or `list`. """
+
+
 class Box(SearchSpace):
     r"""
     Continuous :class:`SearchSpace` representing a :math:`D`-dimensional box in
     :math:`\mathbb{R}^D`. Mathematically it is equivalent to the Cartesian product of :math:`D`
     closed bounded intervals in :math:`\mathbb{R}`.
     """
 
-    def __init__(self, lower: TensorType, upper: TensorType):
-        """
-        :param lower: The lower (inclusive) bounds of the box.
-        :param upper: The upper (inclusive) bounds of the box.
-        :raise ValueError: If ``lower`` and ``upper`` have different shapes. Or if ``upper`` is not
+    def __init__(self, lower: Vector, upper: Vector):
```

Comment:
I've restricted to `tf.Tensor` for simplicity. The float type check doesn't work with numpy arrays